darts/utils/data/tabularization.py

```diff
@@ -161,30 +161,115 @@ def _create_lagged_data(
+    min_target_lag: int,
+    output_chunk_length: int,
+    min_past_cov_lag: Optional[int] = None,
+    min_future_cov_lag: Optional[int] = None,
+    max_future_cov_lag: Optional[int] = None,
+    past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+    future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+    max_samples_per_ts: Optional[int] = None,
+):
     """
-    Add static covariates to the features. Accounts for series with potentially different static covariates
-    by padding with 0 to accomodate for the maximum number of available static_covariates in any of the given
-    series in the sequence. If no static covariates are provided for a given series, its corresponding features
-    are padded with 0.
+    Add static covariates to the features' table for RegressionModels.
+    Accounts for series with potentially different static covariates by padding with 0 to accomodate for the maximum
+    number of available static_covariates in any of the given series in the sequence.
+    If no static covariates are provided for a given series, its corresponding features are padded with 0.
+    Accounts for the case where the model is trained with series with static covariates and then used to predict
+    on series without static covariates by padding with 0 the corresponding features of the series without
+    static covariates.
+
+    Parameters
+    ----------
+    model
+        The regression model. Should be an instance of darts.models.RegressionModel.
+    features
+        The features' numpy array to which the static covariates will be added.
+    target_series
+        The target series from which to read the static covariates.
+    min_target_lag
+        The minimum past lag of the target series.
+    output_chunk_length
+        The model's output chunk length.
+    min_past_cov_lag
+        The minimum past lag of the past covariates.
+    min_future_cov_lag
+        The minimum past lag of the future covariates.
+    max_future_cov_lag
+        The maximum future lag of the future covariates.
+    max_samples_per_ts
+        Optionally, the maximum number of samples to be drawn for training or inference.
+        It is set to 1 for inference (i.e., auto-regressively predict one step at a time, hence generate one
+        feature vector at a time).
     """
 
-    series = series2seq(series)
-    reps = features.shape[0] // len(series)
+    model = model.model
+
+    target_series = series2seq(target_series)
+    past_covariates = series2seq(past_covariates)
+    future_covariates = series2seq(future_covariates)
+
+    max_past_lag = max(
+        [
+            abs(v)
+            for v in [min_target_lag, min_past_cov_lag, min_future_cov_lag]
+            if v is not None
+        ]
+    )
+
     # collect static covariates info
-    map = {"covs_width": [], "values": []}
-    for ts in series:
+    map = {"covs_width": [], "values": [], "reps": []}
```

Comment:
This is not a good name as it overwrites the builtin `map`. Could you maybe rename to `map_` ?

Reply:
oops!